osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -139,6 +154,15 @@ private static void adjustScaleFromAnchor(ref Vector2 scale, Anchor reference)
             // reverse the scale direction if dragging from top or left.
             if ((reference & Anchor.x0) > 0) scale.X = -scale.X;
             if ((reference & Anchor.y0) > 0) scale.Y = -scale.Y;
+
+            // for now aspect lock scale adjustments that occur at corners.
+            if (!reference.HasFlagFast(Anchor.x1) && !reference.HasFlagFast(Anchor.y1))
+            {
+                if (reference.HasFlagFast(Anchor.x0) || reference.HasFlagFast(Anchor.x2))
+                    scale.Y = scale.X;
+                else
+                    scale.X = scale.Y;
```

Comment:
What was the intention here? Asking because the `else` branch seems effectively dead here. I've breakpointed on it and it seems to be unreachable. To explain what I mean:

```
    x0  x1  x2
    |   |   |
y0 -O---O---O-
    |   |   |
y1 -O---+---O-
    |   |   |
y2 -O---O---O-
    |   |   |
```

The `O`s indicate where the reference anchors are on a selection box. The first conditional eliminates the middle ones, which makes sense. But after excluding them from further deliberations (marking via `X`):

```
    x0  x1  x2
    |   |   |
y0 -O---X---O-
    |   |   |
y1 -X---+---X-
    |   |   |
y2 -O---X---O-
    |   |   |
```

all the reference anchors left have `Anchor.x0` or `Anchor.x2`...

Reply:
i'm not actually sure, but it felt okay. this code will likely be removed/replaced within the next week, so just killing that line is fine for now.

Reply:
I *think* what may have been wanted is

```csharp
if (scale.X != 0 && scale.Y != 0)
    scale.X = scale.Y = Math.Max(scale.X, scale.Y);
else if (scale.X != 0)
    scale.Y = scale.X;
else if (scale.Y != 0)
    scale.X = scale.Y;
```

The misleading part about all this is that `scale` is all deltas, not the actual full final scale.

Reply:
I tried that initially but it doesn't work amazingly because as you say, it's arriving as deltas and therefore not in a stable state.

mThats why i chose to leave all the ￼decisions based on the dragged anchor alone.

Reply:
Fair enough. I'll just kill the `else` branch and leave a TODO for now, then...